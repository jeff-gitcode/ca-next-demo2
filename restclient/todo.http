@host=http://127.0.0.1:8090


### GET todo records
GET {{host}}/api/collections/todo/records?page=1&perPage=10 HTTP/1.1
content-type: application/json

### GET todo record
GET {{host}}/api/collections/todo/records/xucm9od3kwqq7zq HTTP/1.1
content-type: application/json

### CREATE todo record
POST {{host}}/api/collections/todo/records HTTP/1.1
accept: text/plain
content-type: application/json

{
  "title": "Admin1",
  "content": ""
}

### Update todo record
PATCH {{host}}/api/collections/todo/records/xucm9od3kwqq7zq HTTP/1.1
accept: text/plain
content-type: application/json

{
  "title": "Admin updated",
  "content": "content1"
}

### DELETE todo record
DELETE  {{host}}/api/collections/todo/records/szfhv770n23b80j HTTP/1.1
accept: text/plain
content-type: application/json


### GET todo records from nextjs
GET http://127.0.0.1:3000/api/todos HTTP/1.1
content-type: application/json

### GET todo record
GET http://127.0.0.1:3000/api/todos/xucm9od3kwqq7zq HTTP/1.1
content-type: application/json

### CREATE todo record
POST http://127.0.0.1:3000/api/todos HTTP/1.1
accept: text/plain
content-type: application/json

{
    "title": "Admin"
}

### UPDATE todo record
PATCH http://127.0.0.1:3000/api/todos/xucm9od3kwqq7zq HTTP/1.1
accept: text/plain
content-type: application/json

{
    "title": "Admin UPDATED"
}

### DELETE todo record
DELETE  http://127.0.0.1:3000/api/todos/3cyet2vmpwj7h2a HTTP/1.1
accept: text/plain
content-type: application/json

### LoginUserFunction (User)
POST {{host}}/api/LoginUserFunction HTTP/1.1
accept: text/plain
content-type: application/json

{
  "name": "Jeff",
  "role": "User",
  "email": "user@email.com",
  "password": "password"
}

### LoginUserFunction (Admin)
POST {{host}}/api/LoginUserFunction HTTP/1.1
accept: text/plain
content-type: application/json

{
  "name": "Admin",
  "role": "Admin",
  "email": "admin@email.com",
  "password": "password"
}

### SearchUserFunction
POST {{host}}/api/SearchUserFunction HTTP/1.1
Authorization: Bearer {{token}}
accept: text/plain
content-type: application/json

{
  "name": "John",
  "page": 1,
  "size": 1
}
