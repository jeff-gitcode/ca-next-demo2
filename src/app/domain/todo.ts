export type Todo = {
  readonly id: string;
  readonly title: string;
};
