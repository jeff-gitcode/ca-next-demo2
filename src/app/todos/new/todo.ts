import z from "zod";

export const todoSchema = z.object({
    id: z.string(),
    title: z.string().min(1),
});

export type FormValue = z.infer<typeof todoSchema>;