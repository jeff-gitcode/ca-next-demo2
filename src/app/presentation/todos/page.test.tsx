import { render, screen } from "@testing-library/react";
import React from "react";

import TodoPage from "./page";
import * as UseTodoController from "@/app/application/hooks/use.todo.controller";
// import * as ApplicationContext from "@/app/application/hooks/app.context";



describe("TodoPage", () => {
    test("should render", () => {
        // Arrange
        // const mockTodoListUseCase = jest.fn();
        // const mockDeleteTodoUseCase = jest.fn();

        const data = [
            {
                id: "1",
                title: "title",
                description: "description",
                status: "status",
                created_at: "created_at",
                updated_at: "updated_at",
            },
        ];
        const isLoading = false;
        const error = undefined;
        const deleteData = undefined;
        const deleteTodo = jest.fn();
        const isDeleting = false;
        // const contextValues = {
        //     container: null,
        //     todoListUseCase: mockTodoListUseCase,
        //     todoUseCase: jest.fn(),
        //     createTodoUseCase: jest.fn(),
        //     updateTodoUseCase: jest.fn(),
        //     deleteTodoUseCase: mockDeleteTodoUseCase,
        // };

        // jest.mock("@/app/application/hooks/app.context");
        // const useContextSpy = jest.spyOn(ApplicationContext, "useAppContext").mockImplementation(() => contextValues);

        // const todoListUseCaseSpy = jest.spyOn(UseTodoController, "useTodoListUseCase").mockReturnValue({
        //     data,
        //     isLoading,
        //     error,
        // });
        jest.mock("@/app/application/hooks/use.todo.controller", () => ({
            useTodoListUseCase: jest.fn().mockReturnValue({ data, isLoading, error }),
            useDeleteTodoUseCase: jest.fn().mockReturnValue({ deleteData, deleteTodo, isDeleting }),
        }));

        // const deleteTodoUseCaseSpy = jest.spyOn(UseTodoController, "useDeleteTodoUseCase").mockReturnValue({
        //     deleteData,
        //     deleteTodo,
        //     isDeleting,
        // });
        const consoleSpy = jest.spyOn(console, "log").mockImplementation();
        render(<TodoPage />);
        screen.debug();
        // Act
        // Assert
        // expect(screen).toMatchSnapshot();
        // expect(useContextSpy).toHaveBeenCalled();
        // expect(todoListUseCaseSpy).toHaveBeenCalled();
        // expect(deleteTodoUseCaseSpy).toHaveBeenCalled();
        // expect(consoleSpy).toHaveBeenCalledWith(data);
    });


    // xtest("should render loading", () => {
    //     // Arrange
    //     const todoListUseCase = jest.fn();
    //     const deleteTodoUseCase = jest.fn();
    //     const data = undefined;
    //     const isLoading = true;
    //     const error = undefined;
    //     const deleteData = undefined;
    //     const deleteTodo = jest.fn();
    //     const isDeleting = false;
    //     const useContextSpy = jest.spyOn(React, "useContext").mockReturnValue({
    //         todoListUseCase,
    //         deleteTodoUseCase,
    //     });
    //     const todoListUseCaseSpy = jest.spyOn(useTodoController, "useTodoListUseCase").mockReturnValue({
    //         data,
    //         isLoading,
    //         error,
    //     });
    //     const deleteTodoUseCaseSpy = jest.spyOn(useTodoController, "useDeleteTodoUseCase").mockReturnValue({
    //         deleteData,
    //         deleteTodo,
    //         isDeleting,
    //     });
    //     const consoleSpy = jest.spyOn(console, "log").mockImplementation();
    //     const wrapper = render(<TodoPage />);
    //     // Act
    //     // Assert
    //     expect(wrapper).toMatchSnapshot();
    //     expect(useContextSpy).toHaveBeenCalled();
    //     expect(todoListUseCaseSpy).toHaveBeenCalled();
    //     expect(deleteTodoUseCaseSpy).toHaveBeenCalled();
    //     expect(consoleSpy).not.toHaveBeenCalled();
    // }
    // );

    // xtest("should render error", () => {
    //     // Arrange
    //     const todoListUseCase = jest.fn();
    //     const deleteTodoUseCase = jest.fn();
    //     const data = undefined;
    //     const isLoading = false;
    //     const error = "error";
    //     const deleteData = undefined;
    //     const deleteTodo = jest.fn();
    //     const isDeleting = false;
    //     const useContextSpy = jest.spyOn(React, "useContext").mockReturnValue({
    //         todoListUseCase,
    //         deleteTodoUseCase,
    //     });
    //     const todoListUseCaseSpy = jest.spyOn(useTodoController, "useTodoListUseCase").mockReturnValue({
    //         data,
    //         isLoading,
    //         error,
    //     });
    //     const deleteTodoUseCaseSpy = jest.spyOn(useTodoController, "useDeleteTodoUseCase").mockReturnValue({
    //         deleteData,
    //         deleteTodo,
    //         isDeleting,
    //     });
    //     const consoleSpy = jest.spyOn(console, "log").mockImplementation();
    //     const wrapper = render(<TodoPage />);
    //     // Act
    //     // Assert
    //     expect(wrapper).toMatchSnapshot();
    //     expect(useContextSpy).toHaveBeenCalled();
    //     expect(todoListUseCaseSpy).toHaveBeenCalled();
    //     expect(deleteTodoUseCaseSpy).toHaveBeenCalled();
    //     expect(consoleSpy).not.toHaveBeenCalled();
    // });

    // xtest("should call deleteTodo", () => {
    //     // Arrange
    //     const todoListUseCase = jest.fn();
    //     const deleteTodoUseCase = jest.fn();
    //     const data = [
    //         {
    //             id: "1",
    //             title: "title",
    //             description: "description",
    //             status: "status",
    //             created_at: "created_at",
    //             updated_at: "updated_at",
    //         },
    //     ];
    //     const isLoading = false;
    //     const error = undefined;
    //     const deleteData = undefined;
    //     const deleteTodo = jest.fn();
    //     const isDeleting = false;
    //     const useContextSpy = jest.spyOn(React, "useContext").mockReturnValue({
    //         todoListUseCase,
    //         deleteTodoUseCase,
    //     });
    //     const todoListUseCaseSpy = jest.spyOn(useTodoController, "useTodoListUseCase").mockReturnValue({
    //         data,
    //         isLoading,
    //         error,
    //     });
    //     const deleteTodoUseCaseSpy = jest.spyOn(useTodoController, "useDeleteTodoUseCase").mockReturnValue({
    //         deleteData,
    //         deleteTodo,
    //         isDeleting,
    //     });
    //     const consoleSpy = jest.spyOn(console, "log").mockImplementation();
    //     const wrapper = render(<TodoPage />);
    //     // Act
    //     wrapper.find("button").simulate("click");
    //     // Assert
    //     expect(deleteTodo).toHaveBeenCalled({ queryParams: { id: "1" } });
    // });

    // xtest("should call deleteTodo", () => {
    //     // Arrange
    //     const todoListUseCase = jest.fn();
    //     const deleteTodoUseCase = jest.fn();
    //     const data = [
    //         {
    //             id: "1",
    //             title: "title",
    //             description: "description",
    //             status: "status",
    //             created_at: "created_at",
    //             updated_at: "updated_at",
    //         },
    //     ];
    //     const isLoading = false;
    //     const error = undefined;
    //     const deleteData = undefined;
    //     const deleteTodo = jest.fn();
    //     const isDeleting = false;
    //     const useContextSpy = jest.spyOn(React, "useContext").mockReturnValue({
    //         todoListUseCase,
    //         deleteTodoUseCase,
    //     });
    //     const todoListUseCaseSpy = jest.spyOn(useTodoController, "useTodoListUseCase").mockReturnValue({
    //         data,
    //         isLoading,
    //         error,
    //     });
    //     const deleteTodoUseCaseSpy = jest.spyOn(useTodoController, "useDeleteTodoUseCase").mockReturnValue({
    //         deleteData,
    //         deleteTodo,
    //         isDeleting,
    //     });
    //     const consoleSpy = jest.spyOn(console, "log").mockImplementation();
    //     const wrapper = render(<TodoPage />);
    //     // Act
    //     wrapper.find("button").simulate("click");
    //     // Assert
    //     expect(deleteTodo).toHaveBeenCalledWith({ queryParams: { id: "1" } });
    // });
}
);
