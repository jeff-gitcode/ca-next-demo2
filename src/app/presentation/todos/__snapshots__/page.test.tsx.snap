// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoPage should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-3xl mx-auto"
      >
        <header
          class="p-6 border-b flex items-center justify-between bg-blue-500 rounded-lg"
        >
          <a
            class="text-2xl font-bold"
            href="/todos"
          >
            Todos
          </a>
          <a
            class="text-2xl font-bold bg-green-500 grid place-items-center px-4 py-2 rounded-full"
            href="/presentation/todos/new"
          >
            Create
          </a>
        </header>
        <main
          class="p-4 text-large"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-3xl mx-auto"
    >
      <header
        class="p-6 border-b flex items-center justify-between bg-blue-500 rounded-lg"
      >
        <a
          class="text-2xl font-bold"
          href="/todos"
        >
          Todos
        </a>
        <a
          class="text-2xl font-bold bg-green-500 grid place-items-center px-4 py-2 rounded-full"
          href="/presentation/todos/new"
        >
          Create
        </a>
      </header>
      <main
        class="p-4 text-large"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
